File: ruby/clock/clock.rb
class Clock
  def self.at(hour, minutes)
    Clock.new(hour, minutes)
  end

  def initialize(hour=0, minutes=0)
    @internal_minutes = hour * 60 + minutes
  end

  def to_s
    format '%02d:%02d', hours, minutes
  end

  def +(minutes)
    @internal_minutes += minutes
    self
  end

  def ==(new_time)
    to_s == new_time.to_s
  end

  private

  def hours
    @internal_minutes / 60 % 24
  end

  def minutes
    @internal_minutes % 60
  end
end

File: go/clock/clock.go
package clock

import "fmt"

const testVersion = 4

type Clock int

func New(hour, minute int) Clock {
	time := (hour*60 + minute) % (60 * 24)
	if time < 0 {
		time += 60 * 24
	}
	return Clock(time)
}

func (c Clock) String() string {
	return fmt.Sprintf("%02d:%02d", c/60, c%60)
}

func (c Clock) Add(minutes int) Clock {
	return New(0, int(c)+minutes)
}
